solutions/search/semantic-search/using-openai-compatible-models.md

```diff
@@ -0,0 +1,185 @@
+   ```
+
+## Expose the endpoint using ngrok
+
+Since the created endpoint only works locally, it cannot be accessed from external services (for example, your Elastic Cloud instance).
+[ngrok](https://ngrok.com/) enables you to expose a local port with a public URL.
+
+1. Create an ngrok account and follow the [official setup guide](https://dashboard.ngrok.com/get-started/setup).
+2. After installing and configuring the ngrok agent, expose the Ollama port by running:
+   ```shell
+   ngrok http 11434 --host-header="localhost:11434"
+   ```
+   The command returns a public link that works as long as ngrok and the Ollama server are running locally:
+   ```shell
+   Session Status                online                                                                                                                                                                              
+   Account                       xxxx@yourEmailProvider.com (Plan: Free)                                                                                                                                             
+   Version                       3.18.4                                                                                                                                                                              
+   Region                        United States (us)                                                                                                                                                                  
+   Latency                       561ms                                                                                                                                                                               
+   Web Interface                 http://127.0.0.1:4040                                                                                                                                                               
+   Forwarding                    https://your-ngrok-endpoint.ngrok-free.app -> http://localhost:11434                                                                                                                   
+   
+   
+   Connections                   ttl     opn     rt1     rt5     p50     p90                                                                                                                                         
+                                 0       0       0.00    0.00    0.00    0.00
+   ```
+
+3. Copy the ngrok-generated URL from the `Forwarding` line.
+4. Test the endpoint again using the new URL:
+   ```shell
+    curl https://your-ngrok-endpoint.ngrok-free.app/api/generate -d '{
+   "model": "llama3.2",
+   "prompt": "What is the capital of France?"
+   }'
+   ```
+   The response should be similar to the previous one.
+
+## Connecting the local LLM to Playground
```

Comment:
I wonder if this should be a snippet that we can re-use here and in the Playground docs

Right now you don't have any cross-reference in the Playground docs which means this info is kind of hidden.